### PR TITLE
vscode: added audit at ci

### PR DESCRIPTION
**Pull request by Veetaha**

### CODE REVIEW

**Comment by bjorn3** [comment]
Alsl add `cargo audit`?

**Comment by Veetaha** [comment]
Sure, I am looking at how to run audit for our vscode extension, since it is located in a subfolder, which actions do not support (...

**Comment by matklad** [comment]
r-

I believe we should fix this by not having typescript dependencies.

Additionally, GitHub actions do not lock third-party actions, which is a major security vulnerability: 3rd party can update their action code and then publish a malicious package using our credentials.

**Comment by matklad** [comment]
cargo audit would actually be a good thing to have though!

**Comment by Veetaha** [comment]
what about just npm audit?

**Comment by Veetaha** [comment]
Btw we can lock the version of an action by specifing the full version `MAJOR.MINOR.PATCH`

**Comment by matklad** [comment]
:+1: on npm audit!

**Comment by Veetaha** [comment]
done, though I think the set of dependencies we use now is fully sound (at least I can tell for node-fetch (it has 0 deps, 13M of downloads and fully sound code)

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -94,9 +94,15 @@ jobs:
         with:
           node-version: 12.x
 
+      - name: Audit vulnerabilities
+        run: npm audit
```
Comment (matklad) [inline .github/workflows/ci.yaml]: Hm, should we run `npm audit` *after* `npm ci`?

Comment (Veetaha) [reply .github/workflows/ci.yaml]: There is nothing about that in [docs](https://docs.npmjs.com/cli/audit) tough I guess we should move it after `npm ci` to be on the safe side

**Comment by matklad** [comment]
Fear not, this is fixed in new version: `@types/node-fetch 2.5.5`  depends
on `form-data 3.0.0` which depends on asynckit and combined-stream.
combined-stream "name" doesn't seem that much different from
[flatmap-stream](
https://snyk.io/blog/malicious-code-found-in-npm-package-event-stream/). I
have no idea why types package would have a dependency.

```
        "@types/node-fetch": {
            "version": "2.5.5",
            "resolved": "
https://registry.npmjs.org/@types/node-fetch/-/node-fetch-2.5.5.tgz",
            "integrity":
"sha512-IWwjsyYjGw+em3xTvWVQi5MgYKbRs0du57klfTaZkv/B24AEQ/p/IopNeqIYNy3EsfHOpg8ieQSDomPcsYMHpA==",
            "dev": true,
            "requires": {
                "@types/node": "*",
                "form-data": "^3.0.0"
            }
        },
```

**Comment by Veetaha** [comment]
@matklad, please don't be paranoic, [`@types/*`](https://github.com/DefinitelyTyped/DefinitelyTyped) are supported by Microsoft and contain nothing but type declarations.

**Comment by Veetaha** [comment]
https://github.com/DefinitelyTyped/DefinitelyTyped/commit/7da30d6637d3d347d24cc44f394844bc34a1ba1c

**Pull request merged**